lib/components/user/new-favorite-location.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+import {
+  FormGroup,
+  InputGroup
+} from 'react-bootstrap'
+
+import Icon from '../narrative/icon'
+import {
+  FavoriteLocationField,
+  FIELD_HEIGHT_PX,
+  FixedLocationType,
+  InvisibleAddon,
+  LocationWrapper,
+  makeFavoriteLocation
+} from './favorite-location-controls'
+
+const blankNewLocationState = {
+  newLocation: null
+}
+
+/**
+ * Component that lets the user pick a new favorite location.
+ */
+class NewFavoriteLocation extends Component {
```

Comment:
It's probably not very clear in the mockups, but this component is intended to be a button (as were the other items in the list). When clicked, it opens a modal where you can set the name and search for geocode results. The main reason for this is so that the mobile experience is not degraded (things don't work well in mobile with the dropdown results, which is why we added the full screen view when setting the from/to location when planning a trip). We should probably discuss more on a call next week to get on the same page.